ui/shared/components/form/Inputs.jsx

```diff
@@ -266,21 +266,23 @@ const InlineFormGroup = styled(Form.Group).attrs({ inline: true })`
 
 export const CheckboxGroup = React.memo((props) => {
   const { value, options, label, groupLabel, onChange, ...baseProps } = props
+  const otherValue = value.filter(val => !options.find(opt => opt.value === val))
+  const realLength = value.length - otherValue.length
```

Comment:
I don't think you need to compute the list of values that are and are not in the list of options, its much cleaner to compute the selected options, and then only compute the updated lists in the onChange method itself, as theres no need to precompute values that may never be used
```
const numSelected = options.filter(opt => value.includes(opt)).length
...
checked={numSelected === options.length}
indeterminate={numSelected > 0 && numSelected < options.length}
...
// select all
onChange([...Set([...value, ...options.map(option => option.value)])])
//deselect all
onChange(value.filter(val => !options.find(opt => opt.value === val)))
```